src/common/hook/useRevokeQueue/useRevokeQueue.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState } from "react";
```

Comment:
concerning this file itself, why not make `usePublishQueue` generic (something like useQueue) so that it could be used with revoke and publish depending on the type passed in?

Reply:
yeah, understand, will update it accordingly